pkg/apis/serving/v1alpha2/framework_tensorflow.go

```diff
@@ -54,6 +54,18 @@ func (t *TensorflowSpec) GetContainer(modelName string, parallelism int, config
 		Command:   []string{TensorflowEntrypointCommand},
 		Resources: t.Resources,
 		Args:      arguments,
+		LivenessProbe: &v1.Probe{
+			Handler: v1.Handler{
+				HTTPGet: &v1.HTTPGetAction{
+					Path: "/v1/models/" + modelName,
```

Comment:
Is it possible that if a model load takes ~10 minutes, liveness could kill the pod forever?

Would be great to support healthz https://github.com/tensorflow/serving/issues/671. In the multimodel case, there may not be a model server, but the server is happy and healthy.

Reply:
Correct. As per the PR description, you cannot currently use KFServing with models that take longer than 2minutes to load. The question is now that KNative is making it configurable, is 180s going to bite anyone, or do we need to make it configurable?

Reply:
@salanki @ellistarn In v1beta1 user can configure the custom liveness probe if they are not happy with the defaults since we allow container fields even for prepackaged model servers.  I suggest here do not use `/v1/models` because that won't work with multi model case @yuzliu, does tensorflow have server liveness probe endpoint?

Reply:
According to the issue @ellistarn linked, there is no real health endpoint yet. There might be a better REST endpoint to pull that works for multi model as well though. Looking for input from someone with multi model experience.

Reply:
Hi, I like the idea to add a liveness check, but it should be on the model server level not the model level since we will allow users to load multiple models soon. Can we use it as an opportunity to push tfserving to add a health check endpoint?

Reply:
I think a simple tcp check can also work here for the liveness probe?

Reply:
Do you know if tcp probes work with KNative? It might work, but by checking a model endpoint we can know if TF thinks that model is alive or not.

Reply:
tcp probe should work as their default readiness probe is tcp check. Definitely agree that model endpoint is better, but we will need to check if it is single model case or multi model.

Reply:
+1 to tcp check on TFServing. I've done this in a past life and platform ;).

@salanki can you configure this to 10 minutes? I think having a model take 10 minutes to load is more common than this failure mode. A worthwhile tradeoff?

Also, as per Dan's comment, just because we enable passthrough of probes doesn't mean we won't provide defaults. Because of this weird tensorflow issue, we probably want to give users a sane default (like 10 minutes) and they can override if they need.

Will lgtm with a higher value, but I /approve of this change.

Reply:
This sounds reasonable, will update.

Reply:
Done

Reply:
Is multi model serving coming in the nexts release? Ie. do we need to address it here, or can we merge this as is and figure out how to best handle multi model for next release?

Reply:
MMS won't be in this release, I think we can address it in next release.

Reply:
For MMS case I think we will want tcp check because we do not want to restart server just because one of the models is not in ready state, we may end up with awkward if/else case.

Reply:
I’ll run some tests and we can keep discussing it in another issue for MMS.